core/http/src/raw_str.rs

```diff
@@ -759,7 +759,7 @@ macro_rules! impl_partial {
     )
 }
 
-impl_partial!(RawStr : &RawStr);
+// impl_partial!(RawStr : &RawStr); -> Using the default implementation to match Hash's default implementation
```

Comment:
I don't understand; why are we removing this?

Reply:
This one was to fix Clippy's error [derive_ord_xor_partial_ord](https://rust-lang.github.io/rust-clippy/master/index.html#derive_ord_xor_partial_ord). When I've removed `PartialOrd` and `Ord`, I've got the error [derive_hash_xor_eq](https://rust-lang.github.io/rust-clippy/master/index.html#derive_hash_xor_eq). I'm not fluent enough in Rust to know what is the underlying problem (this is my first pull request in Rust), but my guess is we should do one of the following :
- Comment the impl_partial that override the default implementation (which I did because it is easier, I've tried to explain why inside a comment because I knew it would not be obvious).
- Create an issue to fix `impl_partial` properly.
If you prefer the latter, I will add a TODO on this line.

Reply:
All of the existing code is correct. The lint appears to be wrong; we _do_ derive both `Ord` and `PartialOrd`; these implementations are for different types which we can't derive implementations for. What's more, they do the same thing. So, this one is a false positive.

Reply:
Since the macro `impl_partial` is and will remain private, it is fine to allow this lint. So I'll do it with these lines :
```
#[allow(clippy::derive_ord_xor_partial_ord)] // The default implementation of Eq, Ord and Hash
#[allow(clippy::derive_hash_xor_eq)]         // works well with the custom ones.
```